Lib/fontParts/base/segment.py

```diff
@@ -279,7 +279,15 @@ def _get_onCurve(self):
         """
         Subclasses may override this method.
         """
-        return self.points[-1]
+        value = self.points[-1]
+        if value.type == "offcurve":
+            # create a dummy implied on curve point
```

Comment:
Why not return None in this case?

Reply:
that was my first idea.. but it opens a gigantic can of snowballs changes:
* segmentType normalisation should support None
* setting the segment.type should support None

Tried to keep make the change as small as possible. This reflects what actually happens: a implied point is added (that point has no reference to a contour or glyph).

Reply:
I have no idea what you mean by "This reflects what actually happens". All implied points eventually get added at render time, and this point is not special in any way. You call it a "dummy", and it is indeed redundant. I don't think it should be exposed in the scripting API.

Why should segmentType/segment.type be None? Shouldn't it be "qcurve"?

`segment.onCurve` should be None, and *perhaps* `segment.points[-1]` should be None as well, like the arguments of the qCurveTo segment pen method.

Reply:
changed the logic by not allowing converting such a segment to an other type